code/modules/power/supermatter/supermatter_process.dm

```diff
@@ -138,6 +139,75 @@
 	var/next_sound = round((100 - aggression) * 5)
 	last_accent_sound = world.time + max(SUPERMATTER_ACCENT_SOUND_MIN_COOLDOWN, next_sound)
 
+/obj/machinery/power/supermatter_crystal/proc/animate_the_sm() // Start the loop
+	filters = filter(type="rays", size = clamp(power/30, 0, 125), color = SUPERMATTER_COLOUR, factor = 0.6, density = 12)
```

Comment:
we have filter helpers, you should use them for this (not sure what they'll do to the animate tho now that I think of it)

Reply:
where? I know not of them

Reply:
search for `add_filter` you'll see what I mean

Reply:
So I tried this, it didn't seem to behave how I wanted it to.

Reply:
elaborate?